apps/src/templates/teacherDashboard/EditSectionForm.jsx

```diff
@@ -130,6 +139,49 @@ class EditSectionForm extends Component {
       isNewSection
     } = this.props;
 
+    /**
+    OAuth login types can not be changed.
+    Personal email login type can be changed to word OR picture login type.
+    Picture login type can be changed to word login type.
+    Word login type can be changed to picture login type.
+    **/
+    const changableLoginTypes = [
+      SectionLoginType.word,
+      SectionLoginType.picture,
+      SectionLoginType.email
+    ];
+
+    // Sections that do not yet have students can use any login type.
+    var validLoginTypes;
+    switch (true) {
```

Comment:
Im not sure if this is the most readable pattern to use for this segment of code. What you're doing here is associating with a handful of login types a list of loginTypes they can transform into. A very natural way to do this would be with a dictionary, keyed on loginType with the values being the list of logintypes those keys can turn into. 

The advantage of doing this is twofold - shorter code, and also using a less expressive programming language feature to represent what you're trying to do, making the code easier to read since there is less things for the programmer to watch out for when reading it.

```
const sectionsLoginTypeTransforms = {
    SectionLoginType.email : changeableLoginTypes,
    SectionLoginType.picture: [SectionLoginType.word, SectionLoginType.picture],
    SectionLoginType.word: [SectionLoginType.word, SectionLoginType.picture],
    SectionLoginType.clever: [SectionLoginType.clever],
    SectionLoginType.googleClassroom: [SectionLoginTypes.google_classroom]
}

validLoginTypes = section.studentCount === 0? Object.values(SectionLoginType):  sectionsLoginTypeTransforms[section.loginType]
```

Reply:
Conceptually, this makes sense and I'm on board, but I'm fighting an uphill syntax battle.  Is there something I'm missing? 

<img width="948" alt="Screen Shot 2020-04-30 at 9 27 19 PM" src="https://user-images.githubusercontent.com/12300669/80782227-137bf700-8b2a-11ea-82a3-bdf4f73ce424.png">

Reply:
Hm, it appears it isn't valid syntactically to deference an object key when assigning a key to an object. I think that means that what I suggested isn't possible, but this equivilent and very similar code should be:

```
const sectionLoginTypeTransforms = {}
sectionLoginTypeTransforms[SectionLoginType.picture]  = [SectionLoginType.word, SectionLoginType.picture];
...
...
etc
...
```

Reply:
Hm, it appears it isn't valid syntactically to deference an object key when assigning a key to an object. I think that means that what I suggested isn't possible, but this equivilent and very similar code should be (I checked this on my machine before suggesting it this time!):

```
const sectionLoginTypeTransforms = {}

sectionLoginTypeTransforms[SectionLoginType.picture]  = [SectionLoginType.word, SectionLoginType.picture];
...
...
etc
...
```